cmd/crank/beta/validate/manager_test.go

```diff
@@ -230,6 +258,96 @@ func TestConfigurationTypeSupport(t *testing.T) {
 	}
 }
 
+func TestAddDependencies(t *testing.T) {
+	cd1 := static.NewLayer(configDep1Yaml, types.OCILayer)
+	cd2 := static.NewLayer(configDep2Yaml, types.OCILayer)
+	pd1 := static.NewLayer(providerYaml, types.OCILayer)
+	fd1 := static.NewLayer(funcYaml, types.OCILayer)
+
+	fetchMockFunc := func(image string) (*conregv1.Layer, error) {
+		switch image {
+		case "config-dep-1:v1.3.0":
+			return &cd1, nil
+		case "config-dep-2:v1.3.0":
+			return &cd2, nil
+		case "provider-dep-1:v1.3.0":
+			return &pd1, nil
+		case "function-dep-1:v1.3.0":
+			return &fd1, nil
+		default:
+			return nil, fmt.Errorf("unknown image: %s", image)
+		}
+	}
+
+	type args struct {
+		extensions []*unstructured.Unstructured
+		fetchMock  func(image string) (*conregv1.Layer, error)
+	}
+	type want struct {
+		confs int
+		deps  int
+		err   error
+	}
+	cases := map[string]struct {
+		reason string
+		args   args
+		want   want
+	}{
+		"SuccessfulDependenciesAddition": {
+			// config-dep-1
+			//└─►config-dep-2
+			//config-dep-2
+			//├─►provider-dep-1
+			//└─►function-dep-1
+			reason: "All dependencies should be successfully fetched and added",
+			args: args{
+				fetchMock: fetchMockFunc,
+				extensions: []*unstructured.Unstructured{
+					{
+						Object: map[string]interface{}{
+							"apiVersion": "pkg.crossplane.io/v1alpha1",
+							"kind":       "Configuration",
+							"metadata": map[string]interface{}{
+								"name": "config-dep-1",
+							},
+							"spec": map[string]interface{}{
+								"package": "config-dep-1:v1.3.0",
+							},
+						},
+					},
+				},
+			},
+			want: want{
+				confs: 2, // 1 Base Configuration, 1 Child Configuration
+				deps:  4, // 2 Configurations, 1 provider, 1 function
+				err:   nil,
+			},
+		},
+	}
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+			w := &bytes.Buffer{}
+
+			m := NewManager(".crossplane/cache", fs, w)
```

Comment:
does the global cache change from #5839 affect this cache usage in unit tests? i.e. if this will no longer be a relative path, where does it go? 🤔

Reply:
Actually this cache is not even used since we get the yaml files from a MockFetcher. Normally this would create `.crossplane/cache` directory at the current directory and use it as a cache for next calls but we dont use it. I will change this to `""` to prevent misunderstandings.